llvm/test/CodeGen/AMDGPU/build_vector.ll

```diff
@@ -307,13 +307,13 @@ define amdgpu_kernel void @build_v2i32_from_v4i16_shuffle(ptr addrspace(1) %out,
 ; GFX942-LABEL: build_v2i32_from_v4i16_shuffle:
 ; GFX942:       ; %bb.0: ; %entry
 ; GFX942-NEXT:    s_load_dwordx4 s[0:3], s[4:5], 0x24
-; GFX942-NEXT:    v_mov_b32_e32 v2, 0
+; GFX942-NEXT:    v_mov_b32_e32 v0, 0
 ; GFX942-NEXT:    s_waitcnt lgkmcnt(0)
 ; GFX942-NEXT:    s_lshl_b32 s3, s3, 16
 ; GFX942-NEXT:    s_lshl_b32 s2, s2, 16
-; GFX942-NEXT:    v_mov_b32_e32 v0, s2
-; GFX942-NEXT:    v_mov_b32_e32 v1, s3
-; GFX942-NEXT:    global_store_dwordx2 v2, v[0:1], s[0:1]
+; GFX942-NEXT:    v_mov_b32_e32 v2, s2
+; GFX942-NEXT:    v_mov_b32_e32 v3, s3
+; GFX942-NEXT:    global_store_dwordx2 v0, v[2:3], s[0:1]
```

Comment:
> I'm surprised by some of the test changes using few registers, which also seem to be slightly worse

In this case, we have a copy from the scalar result of s_lshl to the vgpr for the data in global_store.

The vreg_64_align2 dest gets inflated to av_64_align2 . With this PR, that register now gets assigned after the vgpr addr. Thus we assign the addr to v0, and  the data to v[2:3] due to align2 -- with a gap in the register file at v1. 

Maybe it makes sense to prioritize align2 in some way, or have some scheme which prioritizes larger avgpr over smaller vgpr. I'm open to ideas, but I think any allocation order we choose will end up with worse allocation in some cases.